bazelisk.py

```diff
@@ -226,9 +227,12 @@ def download_bazel_into_directory(version, is_commit, directory):
     destination_path = os.path.join(directory, bazel_filename)
     if not os.path.exists(destination_path):
         sys.stderr.write("Downloading {}...\n".format(url))
-        with closing(urlopen(url)) as response:
-            with open(destination_path, "wb") as out_file:
-                shutil.copyfileobj(response, out_file)
+        with tempfile.NamedTemporaryFile(prefix="bazelisk", delete=False) as t:
+            with closing(urlopen(url)) as response:
+                shutil.copyfileobj(response, t)
+            t.flush()
+            os.fsync(t.fileno())
```

Comment:
Why is the flush and fsync needed? Closing the handle should be enough, shouldn't it?

Reply:
As I understand it, `fsync` is needed before a rename to guarantee that `os.rename` doesn't create a 0 length file on posix systems. I'm going off the discussion in https://github.com/golang/go/issues/22397#issuecomment-380831736.

Updated this to use `os.rename` instead of `shutil.move` as a move across filesystems would become a copy and not be atomic. To make sure the tempfile is on the same filesystem, changed to create the temp file in the bazelisk cache directory.

Reply:
Ah! That's a good point. :) Thanks!